### PR TITLE
Fixes #9: Allow dynamic injection of values.

### LINKED ISSUE #9: Allow dynamic injection of values
In https://github.com/consolidation/config/issues/12, a request was made to allow arbitrary expansion of environment variables within substitutions. This could be done today with code similar to the following:
```
$array = \Symfony\Component\Yaml\Yaml::parse(file_get_contents("dune.yml"));
$reference_properties = ['env' => $_ENV];
$expanded = \Grasmash\YamlExpander\Expander::expandArrayProperties($array, $reference_properties);
```
In the example above, `${env. MY_ARBITRARY_VAR}` would be substituted with the value of the environment variable `MY_ARBITRARY_VAR`.

However, this implementation is undesirable for a couple of reasons. `$_ENV` might not be populated, depending on the active php.ini settings; if `$_ENV` is available, it might be inefficient to copy every environment variable into the reference properties. The caller could copy only specifically named environment variables into the reference properties, but that would be one more thing that would have to be maintained.

Suggestion:

Modify [Expander:: expandProperty()](https://github.com/grasmash/yaml-expander/blob/master/src/Expander.php#L242:L256) to check for dynamic properties when [a value cannot be found](https://github.com/grasmash/yaml-expander/blob/master/src/Expander.php#L246). This point in the execution could be modified to call another routine that would call `getenv()` when the property name begins with `env.`.

This seems generally useful.

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

File: src/Expander.php [hunk 1]
```diff
@@ -241,6 +241,14 @@ public static function expandPropertyWithReferenceData(
      */
     public static function expandProperty($property_name, $unexpanded_value, $data)
     {
+        if (strpos($property_name, "env.") === 0 &&
+          !$data->has($property_name)) {
+            $env_key = substr($property_name, 4);
+            if (!empty($_ENV) && array_key_exists($env_key, $_ENV)) {
```
Comment (grasmash) [inline src/Expander.php]: Not sure why, but getenv() fails here.

Comment (greg-1-anderson) [reply src/Expander.php]: By fails, do you mean does not work with your test where you set `$_ENV['test']`? Try with getenv here and putenv in the test. getenv is far superior to $_ENV, as the former should in theory always work, whereas $_ENV is not always populated.

Comment (grasmash) [reply src/Expander.php]: > By fails, do you mean does not work with your test where you set $_ENV['test']? Try with getenv here and putenv in the test.

But isn't it problematic that users can't set env vars without putenv()?

Comment (greg-1-anderson) [reply src/Expander.php]: This is exactly as you want it to be. Typically, environment variables will be provided from outside the application, not inside it. As previously mentioned, `$_ENV` is not a reliable way to read environment variables; if you want to set an environment variable from inside a program, you should use putenv. Setting `$_ENV` is relying on a side-effect, as `$_ENV` is just a shadow of the environment system.

Comment (greg-1-anderson) [reply src/Expander.php]: (In other words, I am strongly advocating for `getenv`)

Comment (grasmash) [reply src/Expander.php]: I see. So, I've really been using PHP env vars incorrectly!

I found [this comment](http://php.net/manual/en/function.putenv.php#98540) enlightening:

> putenv/getenv, $_ENV, and phpinfo(INFO_ENVIRONMENT) are three completely distinct environment stores. doing putenv("x=y") does not affect $_ENV; but also doing $_ENV["x"]="y" likewise does not affect getenv("x"). And neither affect what is returned in phpinfo().

**Pull request merged**